dotnet/src/Connectors/Connectors.MistralAI/Client/MistralClient.cs

```diff
@@ -35,27 +40,123 @@ internal sealed class MistralClient
+        var request = new TextEmbeddingRequest(this._modelId, data);
+        var endpoint = new Uri($"{this._endpoint}{this._separator}embeddings");
+        using var httpRequestMessage = this.CreatePost(request, endpoint, this._apiKey, false);
+
+        var response = await this.SendRequestAsync<TextEmbeddingResponse>(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+
+        return response.Data.Select(item => new ReadOnlyMemory<float>(item.Embedding.ToArray())).ToList();
     }
 
     #region private
     private readonly string _modelId;
-    private readonly string? _apiKey;
-    private readonly Uri? _endpoint;
+    private readonly string _apiKey;
+    private readonly Uri _endpoint;
     private readonly string _separator;
     private readonly HttpClient _httpClient;
     private readonly ILogger _logger;
+    private ChatCompletionRequest CreateChatCompletionRequest(string modelId, bool stream, ChatHistory chatHistory, MistralAIPromptExecutionSettings? executionSettings)
+    {
+        var request = new ChatCompletionRequest(modelId)
+        {
+            Stream = stream,
+            Messages = chatHistory.Select(chatMessage => new MistralChatMessage(chatMessage.Role.ToString(), chatMessage.Content!)).ToList(),
+        };
+
+        if (executionSettings is not null)
+        {
+            request.Temperature = executionSettings.Temperature;
+            request.TopP = executionSettings.TopP;
+            request.MaxTokens = executionSettings.MaxTokens;
+            request.SafePrompt = executionSettings.SafePrompt;
+            request.RandomSeed = executionSettings.RandomSeed;
+        }
+
+        return request;
+    }
+
+    private HttpRequestMessage CreatePost(object requestData, Uri endpoint, string apiKey, bool stream)
+    {
+        var httpRequestMessage = HttpRequest.CreatePostRequest(endpoint, requestData);
+        this.SetRequestHeaders(httpRequestMessage, apiKey, stream);
+
+        return httpRequestMessage;
+    }
+
+    private void SetRequestHeaders(HttpRequestMessage request, string apiKey, bool stream)
+    {
+        request.Headers.Add("User-Agent", HttpHeaderConstant.Values.UserAgent);
+        request.Headers.Add(HttpHeaderConstant.Names.SemanticKernelVersion, HttpHeaderConstant.Values.GetAssemblyVersion(this.GetType()));
+        request.Headers.Add("Accept", stream ? "text/event-stream" : "application/json");
+        request.Headers.Add("Authorization", $"Bearer {this._apiKey}");
+        request.Content!.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+    }
+
+    private async Task<T> SendRequestAsync<T>(HttpRequestMessage httpRequestMessage, CancellationToken cancellationToken)
+    {
+        //using var response = await this._httpClient.SendWithSuccessCheckAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+        using var response = await this._httpClient.SendAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
```

Comment:
Should we use `SendWithSuccessCheckAsync`? 
```suggestion
        using var response = await this._httpClient.SendWithSuccessCheckAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
```